song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceImpl.java

```diff
@@ -208,6 +218,63 @@ public void updateAnalysis(
     return analyses;
   }
 
+  @Override
+  public GetAnalysisResponse getAnalysis(
+      @NonNull String studyId,
+      @NonNull Set<String> analysisStates,
+      @NonNull int page,
+      @NonNull int size) {
+    studyService.checkStudyExist(studyId);
+    resolveSelectedAnalysisStates(analysisStates);
+
+    val sort = new Sort(Sort.Direction.ASC, ANALYSIS_ID);
+    val pageRequest = PageRequest.of(page, size, sort);
+
+    val spec =
+        new Specification<Analysis>() {
```

Comment:
I'm curious why is this not a part of the `AnalysisSpecificationBuilder` class :stuck_out_tongue:

Reply:
I suppose I could move the logic to `AnalysisSpecificationBuilder`